### PR TITLE
Explain `Astro.glob()` doesn't accept variables

### PR DESCRIPTION
**Pull request by tinymachine**
I'm seeing a lot of folks in the Discord get tripped up on this ([example][1]), so I thought it would warrant a note in the documentation. Please feel free to edit as necessary.

[1]: https://discord.com/channels/830184174198718474/830184175176122389/974711108877316126

#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [X] New or updated content
- [ ] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!

#### Description

- Added a note to the `Astro.glob()` section.

### CODE REVIEW

**Comment by sarah11918** [comment]
Thank you for this @tinymachine ! This looks like a really helpful example, and we'll get on incorporating this into the docs very soon!

**Comment by tony-sull** [comment]
Not sure what the build error in Netlify is after updating to main 🤔

**Comment by delucis** [comment]
> Not sure what the build error in Netlify is after updating to main 🤔

Looks like it’s trying to use `pnpm` and failing to find it? I noticed the builds were running with `yarn` for some reason so this looks like a change. Has someone changed that over in the Netlify config?

Maybe we need to look at the set up we currently recommend for Netlify + pnpm in the docs (but don’t actually use 😅): https://docs.astro.build/en/guides/deploy/#netlifytoml-file

**Comment by tony-sull** [comment]
Ah yeah the last run was me changing it to use PNPM to see if that was the issue, [this](https://app.netlify.com/sites/astro-docs-2/deploys/6283e0f9603d9d000876b459) was the actual build error that I'm trying to track down

**Comment by tony-sull** [comment]
Sorry for the extra noise while I was debugging a build error in our PR builds!

We actually have a separate [PR](https://github.com/withastro/docs/pull/536) open right now to add a troubleshooting guide that should be ready to merge in soon.  This example would be perfect for that guide, would you mind holding on this PR until the troubleshooting guide is merged so the example can live on that page instead of the API reference?

**Comment by sarah11918** [comment]
UPDATE: Converted this to a draft myself, but not go "live on a farm"... we WILL get back to this!

**Comment by sarah11918** [comment]
@tinymachine Just checking in on this! Are you still interested in adding it to the Troubleshooting page?

**Comment by tinymachine** [comment]
Sorry for delay, @sarah11918! If the functionality of `Astro.glob()` hasn't changed since I submitted the PR, I think this could still be helpful for folks, so I think it's worth adding. Let me know if you think otherwise!

**Comment by delucis** [comment]
Totally! No changes to the API, so this would still be useful to add to our new Troubleshooting page! Thanks @tinymachine.

**Comment by sarah11918** [comment]
I've just removed the draft label, as the other PR was merged so we should be able to tackle the content now.

In the meantime, I know some content related to this was added here:  https://docs.astro.build/en/guides/troubleshooting/#supported-values

I'm sorry, I think a different request for similar content arose, and with this still having been marked as a draft, it escaped my mind that we had similar content already here waiting! So, please take a look at what's currently on our troubleshooting page there and let us know if either that works well enough for you, or whether you'd like to PR an edit to that. Thanks for your patience!

File: src/pages/en/reference/api-reference.md [hunk 1]
```diff
@@ -33,6 +33,14 @@ const posts = await Astro.glob('../pages/post/*.md'); // returns an array of pos
...
 `.glob()` only takes one parameter: a relative URL glob of which local files you'd like to import. It’s asynchronous, and returns an array of the exports from matching files.
 
+> NOTE: `Astro.glob()` is a wrapper of Vite's [`import.meta.glob()`](https://vitejs.dev/guide/features.html#glob-import), so it cannot accept variables as they are not statically analyzable. A workaround is to use a static glob pattern and then filter the results. For example:
+
+```js
+const pathToMyPost = 'src/pages/post/my-post.md'
+const posts = await Astro.glob('../pages/post/*.md');
+const myPost = posts.find(post => post.file.endsWith(pathToMyPost))
+```
+
```
Comment (delucis) [inline src/pages/en/reference/api-reference.md:43]: Hi @tinymachine! Just following up. What do you think about this shorter note linking to our new content on the troubleshooting page?

````suggestion
:::note
`Astro.glob()` is a wrapper of Vite's [`import.meta.glob()`](https://vitejs.dev/guide/features.html#glob-import), so it cannot accept variables as they are not statically analyzable. See [the troubleshooting guide](/en/guides/troubleshooting/#supported-values) for a workaround.
:::
````

**Comment by tinymachine** [comment]
So sorry, @delucis and @sarah11918  -- I drafted a response to @sarah11918's question earlier and neglected to post it! I think your edit makes sense, @delucis, and linking to the existing content on the troubleshooting guide is great.

FWIW I think the [code example](https://docs.astro.build/en/guides/troubleshooting/#supported-values) in the troubleshooting guide isn't quite as simple as it could be -- having part of the `pathToMyFeaturedPost` in the example include the `postSlug` taken from the props kind of complicates the issue in question here, in my opinion. I ended up having to reason a bit about what the use case was in that code sample, which I think is somewhat beside the point. But if it's a common use case, then it makes sense to keep it as-is.

Thanks again for keeping me in the loop -- I really appreciate it!

**Pull request merged**